packages/typespec-ts/test/modularUnit/anonymousModel.spec.ts

```diff
@@ -0,0 +1,973 @@
+    describe("happens at body parameter", async () => {
+      function verifyReturnTypeAsEmpty(
+        operationDetail: string,
+        returnType: string
+      ) {
+        assertEqualContent(
+          operationDetail,
+          `
+        import { TestingContext as Client } from "../rest/index.js";
+        import {
+          StreamableMethod,
+          operationOptionsToRequestParameters,
+        } from "@azure-rest/core-client";
+        export function _readSend(
+          context: Client,
+          options: ReadOptions = { requestOptions: {} }
+        ): StreamableMethod<Read200Response> {
+          return context
+            .path("/")
+            .get({ ...operationOptionsToRequestParameters(options) });
+        }
+        export async function _readDeserialize(result: Read200Response): Promise<${returnType}> {
+          if (result.status !== "200") {
+            throw result.body;
+          }
+          return result.body;
+        }
+        export async function read(
+          context: Client,
+          options: ReadOptions = { requestOptions: {} }
+        ): Promise<${returnType}> {
+          const result = await _readSend(
+            context,
+            options
+          );
+          return _readDeserialize(result);
+        }`,
+          true
+        );
+      }
+      it("should map empty anonymous model({}) => Record<string, any>", async () => {
+        const tspContent = `
+        op read(): {};
+        `;
+        // No models.ts file generated
+        assert.isUndefined(await emitModularModelsFromTypeSpec(tspContent));
+        const operationFiles = await emitModularOperationsFromTypeSpec(
+          tspContent
+        );
+        assert.equal(operationFiles?.length, 1);
+        // Generate the operations.ts file with empty model
+        verifyReturnTypeAsEmpty(
+          operationFiles?.[0]?.getFullText()!,
+          "Record<string, any>"
+        );
+      });
+
+      it("should map empty named model(PublishResult {}) => PublishResult", async () => {
+        const tspContent = `
+        model PublishResult {
+        }
+        op read(): PublishResult;
+        `;
+        const modelFile = await emitModularModelsFromTypeSpec(tspContent);
+        // console.log(modelFile?.getFullText());
+        assertEqualContent(
+          modelFile?.getFullText()!,
+          `
+          export interface PublishResult {}
+        `
+        );
+        const operationFiles = await emitModularOperationsFromTypeSpec(
+          tspContent
+        );
+        assert.ok(operationFiles);
+        assert.equal(operationFiles?.length, 1);
+        // Model name referred in operations.ts
+        verifyReturnTypeAsEmpty(
+          operationFiles?.[0]?.getFullText()!,
+          "PublishResult"
+        );
+      });
+
+      it("should return non-empty anonymous  model({ ... })", async () => {
+        const tspContent = `
+        op read(): { foo?: {bar: string | null}};
```

Comment:
Please notice if the return type is anonymous the inline type would be duplicated in different method.

```typescript


export async function _readDeserialize(
     result: Read200Response
): Promise<{ foo?: { bar: string | null } }>

export async function read(
    context: Client,
    options: ReadOptions = { requestOptions: {} }
): Promise<{ foo?: { bar: string | null } }>
```